server/revision/revision.controller.js

```diff
@@ -15,14 +17,40 @@ function index({ course, query }, res) {
   return Revision.findAll(opts).then(data => res.json({ data }));
 }
 
+function restore({ body: { state, entity }, user }, res) {
+  const opts = { recursive: true, isRestore: true, context: { userId: user.id } };
+  const include = [{ model: User, attributes: ['id', 'email'] }];
+  if (entity === 'ACTIVITY') return restoreActivity(state.id, opts, include, res);
+  return restoreTe(state.id, opts, include, res);
+}
```

Comment:
`restoreActivity` and `restoreTe` are helper methods so it makes more sense to have responses in controller method i.e. this one. This would remove this weird response argument.

```js
const action = entity === 'ACTIVITY' ? restoreActivity : restoreTe;
return action(state.id, opts, include)
  .then(data => res.json({ data });
```

Reply:
Hmm, nice 👍